natlib/svcd.c

```diff
@@ -185,3 +185,68 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+static int svcd_unsubscribe( lua_State *L )
+{
+    // check number of args	
+    if (lua_gettop(L) != 4) return luaL_error(L, "Expected (targetip, svcid, attrid, ivkid)");
+    
+    uint16_t svcid = lua_tonumber(L, 2);
+    uint16_t attrid = lua_tonumber(L, 3);
+    uint16_t ivkid = lua_tonumber(L, 4);
+    // set array to var msg
+    
+    lua_getglobal(L, "SVCD"); //index 5
+    lua_pushstring(L, "msg");
+    storm_array_nc_create(L, 7, ARR_TYPE_UINT8);
```

Comment:
It might be better to push the light function and arguments and then use lua_call to invoke it since we should make no assumption on the C version of the code and also we need to let lua handle the stack management instead of C. 

Untitled